dockerfiles/git.Dockerfile

```diff
@@ -0,0 +1,24 @@
+# syntax=docker/dockerfile:1
+
+ARG GO_VERSION=1.18
+ARG ALPINE_VERSION=3.16
+ARG GIT_VALIDATION_VERSION=v1.1.0
+
+FROM alpine:${ALPINE_VERSION} AS base
+RUN apk add --no-cache git gpg
+
+FROM golang:${GO_VERSION}-alpine AS gitvalidation
+ARG GIT_VALIDATION_VERSION
+RUN --mount=type=cache,target=/root/.cache \
+  --mount=type=cache,target=/go/pkg/mod \
+  GOBIN=/out go install "github.com/vbatts/git-validation@${GIT_VALIDATION_VERSION}"
+
+FROM base AS validate
+ARG COMMIT_RANGE
+RUN if [ -z "$COMMIT_RANGE" ]; then echo "COMMIT_RANGE required" && exit 1; fi
+ENV GIT_CHECK_EXCLUDE="./vendor"
+WORKDIR /src
+RUN --mount=type=bind,target=. \
+  --mount=type=cache,target=/root/.cache \
+  --mount=from=gitvalidation,source=/out/git-validation,target=/usr/bin/git-validation \
+  git-validation -q -range "$COMMIT_RANGE" -run short-subject,dangling-whitespace
```

Comment:
Same nit-pick here;

```dockerfile
RUN --mount=type=bind,target=. \
    --mount=type=cache,target=/root/.cache \
    --mount=from=gitvalidation,source=/out/git-validation,target=/usr/bin/git-validation \
      git-validation -q -range "$COMMIT_RANGE" -run short-subject,dangling-whitespace
```

Also looks like we're a bit inconsistent in using `$VAR` vs `${VAR}`

Reply:
Weird idea; instead of having a separate `RUN` to check for the `COMMIT_RANGE`, would it work if we made the `COMMIT_RANGE` compulsory here? (`"${COMMIT_RANGE?no commit range specified}"`)?

```bash
echo "${COMMIT_RANGE?no commit range specified}"
/bin/sh: COMMIT_RANGE: no commit range specified

echo $?
2
```

Reply:
Doing this in the same run will increase build time as context has to be loaded first as well as installing gitvalidation tool, that's why I do it there.

Reply:
> would it work if we made the `COMMIT_RANGE` compulsory here? (`"${COMMIT_RANGE?no commit range specified}"`)?

does not work as it's only triggered on unset

Reply:
> does not work as it's only triggered on unset

Ah, good call. Thought of that, but forgot to test 😂